### DISCUSSION TITLE
Add Osdutil.getSystemDecimalCharacter()

### DISCUSSION THREAD

**Discussion by AroonPro** [OpenSeizureDetector/Android_Pebble_SD#183] [03_Android App]
Char getSystemDecimalCharacter {
	float oneTenth=1/10
 return SubString((String(oneTenth), 2,3)
}

This function needs to be called in onCreate() of Loge port to fill in the decimal point or to parse the string of European decimal to imperial decimal.

Leaving European decimal in time gives an Exception which even failes immediately without "sticking" in osd exception handler.

@jones139 can you in the meantime handle this in all branches?

**Comment by AroonPro** [comment]
Added in v5.0.0_upstream, follow up v5.0.0a @AroonPro
Closing discussion

**Comment by jones139** [comment]
Hi,
Thanks for reporting this issue - where are you seeing the crash?   This is the first report I have received of a problem like this.   I can see what the proposed getSystemDecimalCharacter() function is doing, but I don't know where we need to use it?
Graham.

**Reply by AroonPro** [reply]
It is a problem on devices with non-imperial decimal notation in the formatting of the textbox from duration in the export screen. It (%2.1d) causes to display 2,0 . If the user leaves the decimal separator as , then the application crashes on Double.Parse().
There just before I convert. Just before that, I can add like
In Constants 
public static const DECIMAL_SEPARATOR = '.' 
And in the export activity or somewhere where needed 
 getDecimalSeparator() =! Constants.DECIMAL_SEPARATOR

**Comment by pmithrandir** [comment]
Hello I have a french device and we use , as a separatorDidn't experience that.Also, I thought there was a specific method in android to get the current decimal separator. Any reason why not using it? ( I'm curious)PierreSent from Android deviceLe 2 mai 2024 05:19, AroonPro ***@***.***> a écrit :
Closed #183 as resolved.

**Reply by jones139** [reply]
Thanks - I thought there were quite a lot of users who were using locales that use ',' as the decimal separator - have you used the 'Export Data' function?   It might be limited to the date range selection code in that.

**Comment by pmithrandir** [comment]
Hi,

You are right, the export doesn't work.
I have just emailed you the crash report.

Pierre

**Comment by jones139** [comment]
Thanks - I have created an issue for it (#185)
